overlord/hookstate/ctlcmd/refresh.go

```diff
@@ -0,0 +1,161 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/overlord/hookstate"
+	"github.com/snapcore/snapd/snap"
+)
+
+type refreshCommand struct {
+	baseCommand
+
+	Pending bool `long:"pending" description:"Show pending refreshes of the calling snap"`
+	// these two options are mutually exclusive
+	Proceed bool `long:"proceed" description:"Proceed with potentially disruptive refreshes"`
+	Hold    bool `long:"hold" description:"Do not proceed with potentially disruptive refreshes"`
+}
+
+var shortRefreshHelp = i18n.G("The refresh command prints pending refreshes and can hold disruptive ones.")
```

Comment:
I think I'd make this `hold back` rather than `hold` just to make sure the meaning is clear.